File: frontend/src/api/modules/static-page.ts
import request from '@/utils/request'

export interface StaticPage {
  id: number
  pageKey: string
  title: string
  content: string
  status: number
  createTime: string
  updateTime: string
}

export function getStaticPageByPageKey(pageKey: string) {
  return request.get<StaticPage>(`/static-pages/${pageKey}`)
}

export function getAdminStaticPages() {
  return request.get<StaticPage[]>('/admin/static-pages')
}

export function getAdminStaticPageById(id: number) {
  return request.get<StaticPage>(`/admin/static-pages/${id}`)
}

export function updateAdminStaticPage(id: number, data: Partial<StaticPage>) {
  return request.put<void>(`/admin/static-pages/${id}`, data)
}

File: sc-moyuan-backend/src/main/java/com/moyuan/entity/StaticPage.java
package com.moyuan.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("static_page")
public class StaticPage {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String pageKey;
    private String title;
    private String content;
    private Integer status;
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}
